src/main/java/pl/piekoszek/gorskimatches/equation/EquationRandomizer.java

```diff
@@ -7,14 +7,23 @@
 @Component
 public class EquationRandomizer {
 
-    private final QuizzesGenerator equationGenerator;
+    private final QuizzesGenerator quizzesGenerator;
 
-    public EquationRandomizer(QuizzesGenerator equationGenerator) {
-        this.equationGenerator = equationGenerator;
+    public EquationRandomizer(QuizzesGenerator quizzesGenerator) {
+        this.quizzesGenerator = quizzesGenerator;
     }
 
+    List<String> quizzes = new ArrayList<>();
     public String randomEquation() {
-        var quizzes = new ArrayList<>(equationGenerator.getAllSolutionsByQuiz().keySet());
+        quizzes.addAll(quizzesGenerator.getAllSolutionsByQuiz().keySet());
```

Comment:
This list is going to grow with every request. We'd rather fill it with quizzes in constructor